src/Commands/TideSearchCommands.php

```diff
@@ -0,0 +1,160 @@
+      }
+      $message = ($not_published || $not_in_index) ? $message . "Logged in audit trail as well." : "Nothing to log.";
+      return $message;
+    }
+    catch (ConsoleException $exception) {
+      throw new \Exception($exception->getMessage());
+    }
+  }
+
+  /**
+   * Helper to log into the audit trail.
+   */
+  public static function auditIntoLog($nids, $ops, $description) {
+    $log = [
+      "type" => "tide_search",
+      "operation" => $ops,
+      "description" => t("The following nodes are %des - %nids", [
+        "%nids" => $nids,
+        "%des" => $description,
+      ]),
+      "ref_numeric" => 1,
+      "ref_char" => "drush sapi-nsc node results",
+    ];
+    // Add the log to the "admin_audit_trail" table.
+    if (function_exists("admin_audit_trail_insert")) {
+      admin_audit_trail_insert($log);
+    }
+  }
+
+  /**
+   * Helper to get node Ids from search index.
+   */
+  public static function getNidsFromSearchIndex($indexId) {
+    $command_helper = new CommandHelper(\Drupal::entityTypeManager(), \Drupal::moduleHandler(), \Drupal::service('event_dispatcher'), 'dt');
+    $es_nids = [];
+    $indexes = $command_helper->loadIndexes([$indexId]);
+    if (empty($indexes[$indexId])) {
+      throw new ConsoleException(t('@index was not found'));
+    }
+    $total = $indexes[$indexId]->getTrackerInstance()->getTotalItemsCount();
+    $no_of_batches = ceil($total / 10000);
+    // If the result set is more than 10000 then run it in batch.
+    if ($no_of_batches > 1) {
+      $nid_starting_point = 0;
+      for ($i = 1; $i <= $no_of_batches; $i++) {
+        try {
+          $query = $indexes[$indexId]->query();
+          $query->range(0, 10000);
+          $query->sort('nid');
+          $query->addCondition('nid', $nid_starting_point, '>');
+          $results = $query->execute();
+        }
+        catch (SearchApiException $e) {
+          throw new ConsoleException($e->getMessage(), 0, $e);
+        }
+        foreach ($results->getResultItems() as $item) {
+          $es_nids[] = self::convertToNodeIds($item);
+        }
+        // The last nid will be the starting point for next batch.
+        $nid_starting_point = end($es_nids);
+      }
+      // Will remove if any duplicate values.
+      $es_nids = array_unique($es_nids);
+      return $es_nids;
+    }
+    // Run in single request.
+    try {
+      $query = $indexes[$indexId]->query();
+      $query->range(0, $total);
+      $results = $query->execute();
+    }
+    catch (SearchApiException $e) {
+      throw new ConsoleException($e->getMessage(), 0, $e);
+    }
+    foreach ($results->getResultItems() as $item) {
+      $es_nids[] = self::convertToNodeIds($item);
+    }
+    $es_nids = array_unique($es_nids);
+    return $es_nids;
+  }
+
+  /**
+   * Helper to convert node ids.
+   */
+  public static function convertToNodeIds($item) {
```

Comment:
hey @MdNadimHossain ,
I think you don't need the function `convertToNodeIds`.
the `$results` is an object that implements `ResultInterface`, so I think you can get the actual result by using `$results->getExtraData('elasticsearch_response')` or `$results->getAllExtraData()` , then loop through to get `nid` from the result.